### PR TITLE
Batch operations complete when the result receivers complete (not only finished)

### PR DESCRIPTION
**Pull request by andreidan**
We encountered some issues where when executing batch updates using the
postgres protocol, the returned row count for the operation would be 0.
This happened because when the operation is complete we signal that the
backend is read for query (using the readyForQuery message).
We used to consider the operation complete after we sent the "commandComplete"
messages for every batch in the operation.
Due to the commandComplete channel transimission being async, sometimes
we sent readyForQuery before commandComplete would actually be sent.

This commit changes the definition of "operation complete" from "all result
receivers finished" to "all result receivers complete" (ie. commandComplete
was delivered)

### CODE REVIEW

File: CHANGES.txt [hunk 1]
```diff
@@ -21,6 +21,9 @@ Changes
 Fixes
 =====
 
+- Fixed an issue where batch operations executed using the postgres protocol 
+  could've returned 0 as row count, even though the operation was successful.
```
Comment (mfussenegger) [inline CHANGES.txt]: would rephrase the last part to "even though the actual row count was different".  as `0` could be correct for a succesfull operation, but would be wrong if a failure happened.

Comment (andreidan) [reply CHANGES.txt]: good point! thanks
